numpy/typing/_dtype_like.py

```diff
@@ -79,3 +124,107 @@ class _SupportsDType(Protocol):
 # therefore not included in the Union defining `DTypeLike`.
 #
 # See https://github.com/numpy/numpy/issues/16891 for more details.
+
+# Aliases for commonly used dtype-like objects.
+# Note that the precision of `np.number` subclasses is ignored herein.
+_DTypeLikeBool = Union[
+    Type[bool],
+    Type[np.bool_],
+    "np.dtype[np.bool_]",
+    "_SupportsDType[np.dtype[np.bool_]]",
+    _BoolCodes,
+]
+_DTypeLikeUInt = Union[
+    Type[np.unsignedinteger],
+    "np.dtype[np.unsignedinteger]",
+    "_SupportsDType[np.dtype[np.unsignedinteger]]",
+    _UInt8Codes,
+    _UInt16Codes,
+    _UInt32Codes,
+    _UInt64Codes,
+    _UByteCodes,
+    _UShortCodes,
+    _UIntCCodes,
+    _UIntPCodes,
+    _UIntCodes,
+    _ULongLongCodes,
+]
+_DTypeLikeInt = Union[
+    Type[int],
+    Type[np.signedinteger],
+    "np.dtype[np.signedinteger]",
+    "_SupportsDType[np.dtype[np.signedinteger]]",
+    _UInt8Codes,
```

Comment:
typo here `Uint` -> `Int

Reply:
Good catch! Fixed as of https://github.com/numpy/numpy/pull/18236/commits/d788788f8233d96e647c64bd4070305146b9ff92.